xwiki-pro-macros-api/src/main/resources/templates/html_displayer/icon/edit.vm

```diff
@@ -0,0 +1,42 @@
+## ---------------------------------------------------------------------------
+## See the NOTICE file distributed with this work for additional
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+
+<input type="text" id="buttonIconPicker" name="icon" />
+
+#set($discard = $xwiki.ssx.use('IconThemesCode.IconPicker'))
+#foreach($iconSetName in $services.icon.iconSetNames)
+  #set($discard = $services.icon.render('wiki', $iconSetName))
+#end
+
+<script>
```

Comment:
I'd rather move the following JavaScript and CSS to separate files in the JAR and load them using ``jsfx`` and ``ssfx`` (see the [Skin Extension Plugin](https://extensions.xwiki.org/xwiki/bin/view/Extension/Skin%20Extension%20Plugin)). Moreover, I'd add a CSS class (``icon-picker``) on the input and a ``data-options`` attribute to hold the JSON config (``{prefix: ''}``) and then use that to find and initialize the icon pickers (too bad the ``IconThemesCode.IconPicker`` doesn't provide this).